db/migrate/20210928154029_create_gemini_connection_for_channel.rb

```diff
@@ -0,0 +1,16 @@
+class CreateGeminiConnectionForChannel < ActiveRecord::Migration[6.1]
+  def change
+    create_table :gemini_connection_for_channels, id: :uuid, default: -> { "uuid_generate_v4()"} do |t|
+      t.belongs_to :gemini_connection, type: :uuid, index: true, null: false
+      t.belongs_to :channel, type: :uuid, index: true, null: false
+
+      t.string :channel_identifier, index: true
+      t.string :recipient_id
+
+      t.timestamps
+    end
```

Comment:
```suggestion
    create_table :gemini_connection_for_channels, id: :uuid, default: -> { "uuid_generate_v4()"} do |t|
      t.belongs_to :gemini_connection, type: :uuid, null: false
      t.belongs_to :channel, type: :uuid, index: true, null: false

      t.string :channel_identifier
      t.string :recipient_id

      t.timestamps
    end
```

You'll have to regenerate the `db/schema.rb` file, but basically we have duplicate indices which adds space/performance overhead. Postgres has partial indices unlike MySQL